src/core/thread/mle.cpp

```diff
@@ -4199,43 +4532,42 @@ const char *Mle::MessageActionToString(MessageAction aAction)
 const char *Mle::MessageTypeToString(MessageType aType)
 {
     static const char *const kMessageTypeStrings[] = {
-        "Advertisement",         // (0)  kTypeAdvertisement
-        "Announce",              // (1)  kTypeAnnounce
-        "Child ID Request",      // (2)  kTypeChildIdRequest
-        "Child ID Request",      // (3)  kTypeChildIdRequestShort
-        "Child ID Response",     // (4)  kTypeChildIdResponse
-        "Child Update Request",  // (5)  kTypeChildUpdateRequestOfParent
-        "Child Update Response", // (6)  kTypeChildUpdateResponseOfParent
-        "Data Request",          // (7)  kTypeDataRequest
-        "Data Response",         // (8)  kTypeDataResponse
-        "Discovery Request",     // (9)  kTypeDiscoveryRequest
-        "Discovery Response",    // (10) kTypeDiscoveryResponse
-        "delayed message",       // (11) kTypeGenericDelayed
-        "UDP",                   // (12) kTypeGenericUdp
-        "Parent Request",        // (13) kTypeParentRequestToRouters
-        "Parent Request",        // (14) kTypeParentRequestToRoutersReeds
-        "Parent Response",       // (15) kTypeParentResponse
+        "Advertisement",           // (0)  kTypeAdvertisement
+        "Announce",                // (1)  kTypeAnnounce
+        "Child ID Request",        // (2)  kTypeChildIdRequest
+        "Child ID Request",        // (3)  kTypeChildIdRequestShort
+        "Child ID Response",       // (4)  kTypeChildIdResponse
+        "Child Update Request",    // (5)  kTypeChildUpdateRequestOfParent
+        "Child Update Response",   // (6)  kTypeChildUpdateResponseOfParent
+        "Data Request",            // (7)  kTypeDataRequest
+        "Data Response",           // (8)  kTypeDataResponse
+        "Discovery Request",       // (9)  kTypeDiscoveryRequest
+        "Discovery Response",      // (10) kTypeDiscoveryResponse
+        "delayed message",         // (11) kTypeGenericDelayed
+        "UDP",                     // (12) kTypeGenericUdp
+        "Parent Request",          // (13) kTypeParentRequestToRouters
+        "Parent Request",          // (14) kTypeParentRequestToRoutersReeds
+        "Parent Response",         // (15) kTypeParentResponse
+        "Link Accept",             // (16) kTypeLinkAccept
```

Comment:
Wrap the 3 itmes with `OPENTHREAD_FTD || OPENTHREAD_MTD_S2S`?

Reply:
It is a disaster to add a new macro here. With current `MessageTypeToString` implementation, a newly added macro will brings new combination of the static_assert. I would like to enhance `MessageTypeToString` first but did not find a good solution for C++ 11 to convert enum to string.

Reply:
Let's leave it this way and revisit this after the fix on master has been merged.

Reply:
Sounds good to me.